activation/handler.go

```diff
@@ -232,6 +255,13 @@ func (h *Handler) validateNonInitialAtx(ctx context.Context, atx *types.Activati
 		return fmt.Errorf("num units %d is greater than previous atx num units %d", atx.NumUnits, prevAtx.NumUnits)
 	}
 
+	if atx.VRFNonce != nil {
```

Comment:
what is the protocol rule here? atx can arbitrarily update its vrf nonce? or the PoST size have to increase in this case? if so, does it need to be verified?

Reply:
> atx can arbitrarily update its vrf nonce?

Yes. A smesher can publish a new VRF Nonce any time, when they do it becomes effective with `atx.TargetEpoch`.

> or the PoST size have to increase in this case?

When PoST size increases the node has 2 options: not publish a new VRFNonce (if the existing one satisfies the difficulty required for the increased PoST size) or publish a new VRFNonce that satisfies the higher difficulty.

> if so, does it need to be verified?

Yes in any case the existing (or new) VRFNonce has to be validated again for the higher difficulty when the PoST size increases. This is not part of this PR/Issue (https://github.com/spacemeshos/go-spacemesh/issues/3962) but will be done in a future PR: https://github.com/spacemeshos/go-spacemesh/issues/3963